### PR TITLE
Add: Collaborators to PDF export

### PR DESCRIPTION
**Pull request by Cervajz**
Fix: Project filter in admin/users/1

### CODE REVIEW

File: app/models/entry/finder.rb [hunk 1]
```diff
@@ -70,6 +70,10 @@ def total
       @total ||= entries.sum(:minutes)
     end
 
+    def collaborators
+      @collaborators ||= @entries.map(&:user_id).uniq.map { |user_id| User.find user_id }
```
Comment (papricek) [inline app/models/entry/finder.rb]: User.find(@entries.pluck(:user_id))

Comment (papricek) [inline app/models/entry/finder.rb]: But this will probably fail if empty array will be passed, maybe

    User.find_by(id: @entries.pluck(:user_id))

Comment (simi) [inline app/models/entry/finder.rb]:
```ruby
def entries
  entries = Entry.where(user_id: 1)
end
...
@collaborators ||= User.where(id: entries)
#=> SELECT "users".* FROM "users"  WHERE "users"."id" IN (SELECT "entries"."id" FROM "entries"  WHERE "entries"."user_id" = 1)
```

File: app/views/entries/index.pdf.slim [hunk 2]
```diff
@@ -11,10 +12,15 @@ table.table.table-condensed
 
       tr.day-summary
         td colspan="2" = formatted_minutes(entries.sum(&:minutes))
+        - if @entry_finder.collaborators.count > 1
```
Comment (papricek) [inline app/views/entries/index.pdf.slim]: @entry_finder.show_colllaborators?

Comment (simi) [inline app/views/entries/index.pdf.slim]: any?

File: app/views/entries/_filter.html.slim [hunk 3]
```diff
@@ -9,7 +9,10 @@ div.box-nav
         = select_tag :user_id, options_from_collection_for_select(@project.users, :id, :username, params[:user_id]), prompt: "All users", class: ' form-control'
     - else
       .project
-        = select_tag :project_id, options_from_collection_for_select(current_user.projects, :id, :name, params[:project_id]), prompt: "All projects", class: ' form-control'
+        - if params[:controller] == 'admin/users'
```
Comment (papricek) [inline app/views/entries/_filter.html.slim]: Lets do this in a helper method.

    def project_select_options
      projects = if params[:controller] == "fsafsa" ? @user.projects : current_user.projects

File: app/views/entries/index.pdf.slim [hunk 4]
```diff
@@ -4,6 +4,7 @@ ul
   li= "Project: #{@entry_finder.project_name}" if @entry_finder.project_name
   li= "Period: #{I18n.l(@entry_finder.from, format: :short)} - #{I18n.l(@entry_finder.to, format: :short)}"
   li= "Total: #{formatted_minutes(@entry_finder.total, legend: true)}"
+  li= "Collaborators: #{@entry_finder.collaborators.map(&:username).join(', ')}"
```
Comment (simi) [inline app/views/entries/index.pdf.slim]: maybe simple helper?

Comment (papricek) [inline app/views/entries/index.pdf.slim]: or entry_finder_presenter - that would also be good place for @entry_finder.show_colllaborators?

File: app/models/entry/finder.rb (hunk 1 above)
Comment (simi) [inline app/models/entry/finder.rb]: @papricek (5 seconds in console):

```ruby
User.find_by(id: [])
#=> SELECT  "users".* FROM "users"  WHERE 1=0 LIMIT 1
```

Comment (papricek) [inline app/models/entry/finder.rb]: josef.professional? => true

File: app/views/entries/_filter.html.slim (hunk 3 above)
Comment (simi) [inline app/views/entries/_filter.html.slim]: http://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html#method-i-current_page-3F

**Comment by papricek** [comment]
:+1:

**Pull request merged**